src/Altinn.Platform/Altinn.Platform.Authorization/IntegrationTests/DelegationsControllerTest.cs

```diff
@@ -243,6 +243,90 @@ public async Task Post_DeleteRules_InvalidInput_BadRequest()
             Assert.Null(actual);
         }
 
+        /// <summary>
+        /// Test case: Calling the POST operation for DeleteRules to perform a deletion of org1/app4 and org1/app3 where the user performing the task is not defined
+        /// Expected: DeleteRules returns status code 500 and no list of rules as one of the policies had no DeletedByUser set (0)
+        /// </summary>
+        /// <summary>
+        /// Scenario:
+        /// Calling the POST operation for DeleteRules to delete rules without giving a DeletedByUserId
+        /// Input:
+        /// List of three rules for delegation of the app org1/app3 and org1/app4 between for a single offeredby/coveredby combination resulting in no policy file beeing updated.
+        /// Expected Result:
+        /// No Rules are deleted and no rules are returned
+        /// Success Criteria:
+        /// DeleteRules returns status code 500 and no deletion is performed
+        /// </summary>
+        [Fact]
+        public async Task Post_DeleteRules_InvalidUserPerformingDeleteRule_BadRequest()
+        {
+            // Arrange
+            Stream dataStream = File.OpenRead("Data/Json/DeleteRules/ReadOrg1App3App4_50001337_20001337_NoDeletedBy.json");
+            StreamContent content = new StreamContent(dataStream);
+            content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+
+            // Act
+            HttpResponseMessage response = await _client.PostAsync("authorization/api/v1/delegations/DeleteRules", content);
+
+            string responseContent = await response.Content.ReadAsStringAsync();
+
+            // Assert
+            Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+            List<Rule> actual = null;
+            try
+            {
+                actual = (List<Rule>)JsonConvert.DeserializeObject(responseContent, typeof(List<Rule>));
+            }
+            catch
+            {
+                // do nothing this is expected
+            }
```

Comment:
Instead of trying something we know will fail, we should attempt to deserialize into a ProblemDetails or whatever expected response. I hope that works, but another option is to read the response as a string and check the content to assert that the expected error message is included.

Reply:
Fixed parsing the response as ValidationProblemDetails and fetch the error message from the Error response and compares this against expected error message.

Reply:
Thank you. Much bette.